src/main/java/tutorease/address/logic/commands/DeleteLessonCommand.java

```diff
@@ -45,12 +46,12 @@ public CommandResult execute(Model model) throws CommandException {
 
         int listIndex = targetIndex.getZeroBased();
 
-        if (listIndex >= model.getLessonScheduleSize()) {
+        if (listIndex >= model.getFilteredLessonList().size()) {
             throw new CommandException(MESSAGE_INVALID_INDEX);
         }
 
-        Lesson lesson = model.getLesson(listIndex);
-        model.deleteLesson(listIndex);
+        Lesson lesson = model.getFilteredLessonList().get(listIndex);
```

Comment:
Maybe we can write something like this, to be parallel to the delete contact command:
Now you are handling this using an exception, so I am also thinking...
```
List<Lesson> lastLessonList = model.getFilteredLessonList();
if (listIndex >= lastShownList.size()) {
            throw new CommandException(MESSAGE_INVALID_LESSON_DISPLAYED_INDEX);
}

Lesson lessonToDelete = lastShownList.get(listIndex);
model.deleteLesson(lessonToDelete);
```

Reply:
Maybe we can create a method to get from filtered lesson list to not violate LOD.